plugins/module_utils/rds.py

```diff
@@ -154,6 +155,8 @@ def wait(client, db_instance_id, waiter_name, extra_retry_codes):
         extra_retry_codes = []
     for attempt_to_wait in range(0, 10):
         try:
+            # This seems to help limit failures from the waiter itself
+            sleep(10)
```

Comment:
If at all possible it would be nice to avoid this; RDS is slow enough on its own!  Would you be able to try debugging what boto is doing when this waiter error happens, since it sounds like you can reproduce it fairly often?  It would be interesting to see if there's some other failure we could capture in the waiter that's getting hidden by something else.
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/boto3.html
I've used it on the past like this:
```
 # DEBUG is noisy. Reading it on the console sucks, so use a file
 import logging
 logging.basicConfig(filename='boto.log')
 # You could use name=botocore, or name=boto3. 99.9% of calls will happen in botocore,
 # but leaving this empty captures everything - just in case
     boto3.set_stream_logger(name='', logging=logging.DEBUG)
     paginator = client.get_paginator('list_resource_record_sets')
     record_sets = paginator.paginate(**params).build_full_result()['ResourceRecordSets']
     return {
```

Reply:
Yep I'm not even 100% sure it was making any difference, I'll remove it ! The tests seem better anyway with this slight adjustment, I think there were some other test specific issues.

And thanks for the example for troubleshooting further 👍🏻